kratos/solving_strategies/convergencecriterias/residual_criteria.h

```diff
@@ -348,7 +348,7 @@ class ResidualCriteria
 
             if (it_dof->IsFree()) {
                 dof_id = it_dof->EquationId();
-                residual_dof_value = b[dof_id];
+                residual_dof_value = TSparseSpace::GetValue(b,dof_id);
```

Comment:
guys, i THINK that this operation is extremely expensive in parallel since it involves communication.

could u guys take a look at this? I am reopening the PR

Reply:
Do we open a new PR?